### PR TITLE
Set up cucumber, rspec for testing and coveralls

### PR DESCRIPTION
**Pull request by marizaswart**
## Description
Setting up the environment for work to be done in a test driven way with cucumber, rspec and coveralls.

### What we have learned
- How to set up cucumber, rspec
- How to set up and monitor all code is tested with coveralls

### CODE REVIEW

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+  config.mock_with :rspec do |mocks|
+
+    mocks.verify_partial_doubles = true
+  end
+
+
+  config.shared_context_metadata_behavior = :apply_to_host_groups
+
+
```
Comment (tochman) [inline spec/spec_helper.rb]: I don't see the ShoudaMatchers setup

Comment (tochman) [reply spec/spec_helper.rb]: We also have some instructions for disabling some of the generators in Rails. Please look into that in the course materials.

**Review by tochman — CHANGES_REQUESTED** [review]
Please see comments and I can't see that cucumber has been set up.

**Comment by marizaswart** [comment]
Fixed the shoulda matchers configuration and disabled generators. Cucumber should be working now.

**Comment by marizaswart** [comment]
Fixed shoulda matchers and checked cucumber and rspec both should be working now

**Review by tochman — DISMISSED** [review]
Please remove `coverage` folder from version control

**Review by diraulo — COMMENTED** [review]
We can merge this once you've removed the `coverage` folder from version control

File: .gitignore [hunk 2]
```diff
@@ -25,5 +25,5 @@
 .idea/vcs.xml
 .idea/jsLibraryMappings.xml
 
-
+/coverage
```
Comment (diraulo) [inline .gitignore]: you don't need the leading slash here

**Review by diraulo — CHANGES_REQUESTED** [review]
you still have files from the coverage directory here.. do the following to remove them

```
git rm -rf coverage
git add .
git commit -m 'remove coverage directory from version control'
```

**Review by diraulo — APPROVED** [review]
LGTM :shipit:

**Pull request merged**